### ISSUE TITLE
String placeholder visible when publisher has 0 ebooks

### ISSUE THREAD

**Issue by Double-A-92** [internetarchive/openlibrary#2483]
On the publisher view, if there are 0 published ebooks, the date placeholder text is not rendered / replaced correctly. 

### Evidence / Screenshot (if possible)
Examples:
- Publisher - 1 work / 0 ebooks $_('published in') 2009 
- Publisher - 2 works / 0 ebooks $_('published between') 1997 & 2003. 

![grafik](https://user-images.githubusercontent.com/2192982/66562772-9de41580-eb5c-11e9-86ac-c1a9722ccf1c.png)

### Relevant url?
- https://openlibrary.org/publishers/AH2F_Jerryngrid
- https://openlibrary.org/publishers/Editions_10-18

### Expectation
Show the correct text ressource.

### Details

- **Logged in (Y/N)?**    No 
- **Browser type/version?**    Reproducible with Firefox and Edge 
- **Operating system?**    Windows 10; German region settings

### Proposal & Constraints
Figure out if the text ressources are just missing, or if there is some error with the rendering engine and fix it.
Issue might be in this file: openlibrary/openlibrary/plugins/openlibrary/js/graphs/plot.js

**Comment by cdrini** [comment]
Good find! The `$_('blah')` syntax (which is used for internationalization) only works inside the html template files; not the JS. So either:
1) Long term solution: find a way to put that text in the html.
2) Short term solution: remove the `$_`. This won't i18n correctly though.

Looks like there's another usages of `$_` inside this file for tooltips. I don't think there is, but do you by any chance know if there's a way to do i18n from the frontend, @jdlrobson ?

**Comment by jdlrobson** [comment]
Not yet. Blocked on getting the code out of html and tracked in #2353

**Comment by Double-A-92** [comment]
I just noticed that what I described is not the main problem there... The "Publishing History" plot completely fails to even load whenever there is more than 0 ebooks. That's why the wrong strings were hidden so well, because the script that prints them fails to load in the normal case.

Also I've researched a bit about the i18n and there doesn't seem to be any straight forward way to deal with JS files in Babel ([#](https://stackoverflow.com/questions/22355953/flask-babel-localized-strings-within-js)). It's probably best to have a separate i18n system in the frontend with static JSON files.

Also is it fine if I implement option 2 for the hacktober shirt? 😅

**Comment by cdrini** [comment]
> Also is it fine if I implement option 2 for the hacktober shirt? 😅

:P Go for it!

**Comment by cdrini** [comment]
> The "Publishing History" plot completely fails to even load whenever there is more than 0 ebooks.

😬 Oh wow, you're right! Would you mind opening another issue for that? This is a sample publisher: https://openlibrary.org/publishers/Tor

**Comment by cdrini** [comment]
> Also I've researched a bit about the i18n and there doesn't seem to be any straight forward way to deal with JS files in Babel (#). It's probably best to have a separate i18n system in the frontend with static JSON files.

Yeah; adding i18n to the JS will be a _much_ bigger issue. So I would actually prefer option 2 :P

**Comment by tfmorris** [comment]
@cdrini @jdlrobson Since my comments on the pull request were ignored because I didn't comment here on the issue, let me state publicly that I find it completely unacceptable that we have a practice of breaking stuff during "refactoring" and then using the breakage as justification for ripping it out. In every engineering organization I've ever known, the term "refactoring" implies unchanged functionality, but improved structure.

**Comment by Double-A-92** [comment]
I hope this is not intented to be the final state... Getting the translation to work in "dynamic" JS involves bigger effort, and probably would replace the existing broken code anyway. Keeping the "$_(..." parts in the code would not have been any benefit at all, except maybe for marking where i18n is needed.

**Comment by jdlrobson** [comment]
This is not intended to be the final state. Once the JS is manageable I will ensure we spend time fixing parts of the infrastructure with an emphasis of i18n.

This code was very brittle and this part of the code needs rewriting not refactoring.
